### DISCUSSION TITLE
WebGL limits texture size to 2048?

### DISCUSSION THREAD

**Discussion by tonyyzy** [gfx-rs/wgpu#2952] [Q&A]
When ask for a texture size larger than 2048 I get `Device::create_texture` error. Just want to check if this is a known limit? Did a repo search but couldn't find where it's defined. And iiuc most browsers support 4096?

**Comment by cwfitzgerald** [comment]
It stems from the gles3 spec, which webgl2 builds off of. https://registry.khronos.org/OpenGL/specs/es/3.0/es_spec_3.0.pdf

**Reply by tonyyzy** [reply]
Thanks for pointing to the spec! I see that `MAX TEXTURE SIZE` is listed as minimum 2048. Does that mean wgpu is being super conservative here?

**Reply by cwfitzgerald** [reply]
To make writing portable code easier, we use the limits that you provide to the device as the limits we validate against. This way you don't run into the issue of violating the limits you asked for because you're adapter supports more. You can, however, ask for as high of limits as the adapter supports - if you do adapter.limits() you'll get the maximum values you can request on the device.
